enclave_manager/avalon_enclave_manager/work_order_processor_manager.py

```diff
@@ -57,9 +60,13 @@ def _process_work_order_sync(self, process_wo_id):
         wo_id = self._kv_helper.csv_match_pop(
             "wo-worker-pending", self._worker_id, process_wo_id)
         if process_wo_id == wo_id:
-            # @TODO : Need to check if the workload_id in the work order
-            # is supported by this worker. Process only if check passes.
-            return self._process_work_order_by_id(wo_id)
+            wo_process_result = self._process_work_order_by_id(wo_id)
+
+            self._kv_helper.remove("wo-processing", wo_id)
+            # Cleanup tables that hold in-progress work orders
```

Comment:
By "in-progress" do you mean work orders which are "processing" ?

Reply:
Yes, work orders that have been picked up from scheduled/pending to be processed but haven't yet finished processing.